server/src/main/java/org/eclipse/lsp/cobol/core/visitor/CobolVisitor.java

```diff
@@ -528,6 +530,23 @@ private NodeType getDataDescriptionNodeType(DataDescriptionEntryFormat1Context c
     return outlineTreeBuilder.build(copybooks.getUsages());
   }
 
+  @Override
+  public List<Node> visitSetUpDownByStatement(SetUpDownByStatementContext ctx) {
+    List<Node> result = new ArrayList<>();
+    Locality setStart = positionMapping.get(ctx.getParent().getStart());
+    List<Locality> receivingFields =
+        ctx.receivingField().stream()
+            .map(ParserRuleContext::getStart)
+            .map(positionMapping::get)
+            .filter(Objects::nonNull)
+            .collect(toList());
+    Locality sendingField = positionMapping.get(ctx.sendingField().getStart());
+    String literal = ofNullable(ctx.sendingField().literal()).map(ParserRuleContext::getText).orElse(null);
+    result.add(new SetUpDownByStatement(setStart, receivingFields, sendingField, literal));
+    result.addAll(visitChildren(ctx));
```

Comment:
This will add the `SetUpDownByStatement` and variable usages on the same level.
But you must add children under the `SetUpDownByStatement`. And also you need to check that setStart is not null...
Probably it will be better to check it separately. I'll try to prepare a PR.
But anyway, build the right structure.